app/controllers/subscription_controller.rb

```diff
@@ -45,7 +45,9 @@ def add
         end
       end
 
-      case tag_selection_more_than_zero?(params[:tid])
+      # The route params does not have :tid, it only accepts the name..
+      # I think the name should be used to check if the user is subscribed to a tag already
+      case tag_selection_more_than_zero?(tag.tid)
```

Comment:
The route params does not have :tid, it only accepts the name.

Reply:
aha, ok - i'm fine with this! Uh, are we sure, as in, can we find if this code was tested by looking at the PR it was added, and skimming the functional and system tests to check? Will we need to modify any other code if we make this change? But, totally on-board with doing this, it makes sense!

Reply:
I did search it in the routes folder and didn't find it, I'll have a look at the test as well to confirm.

Reply:
I've added a test for this, pushing it in a few. I also looked up the PR that was created for this change, no test was written and when I checked the routes folder, an add route with the params :tid was & is not present.